manifests/install.pp

```diff
@@ -33,12 +33,18 @@
     require => Exec["pyenv::checkout ${user}"],
   }
 
-  exec { "pyenv::shrc ${user}":
-    command => "echo 'source ${pyenvrc}' >> ${shrc}",
-    user    => $user,
+  file { $shrc:
```

Comment:
@Sliim upon brief quick review it looks like virtual resources may be a good solution to this--thanks for suggesting. Here's another page that I found helpful in reading up on virtual resources: http://apprize.info/usability/puppet/5.html.